src/section/mod.rs

```diff
@@ -0,0 +1,200 @@
+//! Helpers for adding custom sections to error reports
+use std::fmt::{self, Display, Write};
+
+pub mod help;
+
+#[non_exhaustive]
+#[derive(Debug)]
+pub(crate) enum Order {
+    AfterErrMsgs,
+    AfterBackTrace,
+    SkipEntirely,
+}
+
+/// A custom section for an error report.
+///
+/// # Details
+///
+/// Sections consist of two parts, a header, and an optional body. The header can contain any
+/// number of lines and has no indentation applied to it by default. The body can contain any
+/// number of lines and is always written after the header with indentation inserted before
+/// every line.
```

Comment:
nit: colon

```suggestion
/// Sections consist of two parts, a header: and an optional body. The header can contain any
/// number of lines and has no indentation applied to it by default. The body can contain any
/// number of lines and is always written after the header with indentation inserted before
/// every line.
```

Is the indentation on the body optional or mandatory?

Reply:
the indentation on the body is currently mandatory but you can just put everything in the header to avoid this.